### ISSUE TITLE
`files` pattern not handling `str` and `io.BytesIO`

### ISSUE THREAD

**Issue by pierremonico** [lundberg/respx#261]
When using the `files` parameter in a helper (http) function, both `str` and `io.BytesIO` payloads are not deemed equal and raise an error.

# Input

## `str` payload
```py
@respx.mock
def test_file_str_payload():
    FILE_DATA = {"upload": ("image/png", "str", "image/png")}

    respx.patch("https://api.example.com/endpoint/123", files=FILE_DATA)

    httpx.patch("https://api.example.com/endpoint/123", files=FILE_DATA)

```

## `io.BytesIO` payload
```py
@respx.mock
def test_file_bytesio_payload():
    FILE_DATA = {
        "upload": ("image/png", io.BytesIO(b"some image content"), "image/png")
    }

    respx.patch("https://api.example.com/endpoint/123", files=FILE_DATA)

    httpx.patch("https://api.example.com/endpoint/123", files=FILE_DATA)
```

# Output

## Error raised
```bash
raise AllMockedAssertionError(f"RESPX: {request!r} not mocked!")
respx.models.AllMockedAssertionError: RESPX: <Request('PATCH', 'https://api.example.com/endpoint/123')> not mocked!
```

**Comment by pierremonico** [comment]
Addressed by #260 but wanted to provide clearer context.

**Comment by lundberg** [comment]
Thanks @pierremonico.

If you haven't already, please try #260 on your own code to rule out any new bugs.

**Comment by pierremonico** [comment]
I've tested #260 within my codebase and all my expected tests worked. I've also added one containing a `str` payload and that worked too. 

For reference, I've installed the fix from #260 within my environment running the following commands at the root of my project (with the virtual environment in which `respx` was currently installed activated):
```bash 
git clone git@github.com:lundberg/respx.git
cd respx 
git checkout fix/files-pattern-io
pip install .
cd ..
rm -rf respx
```

I believe you can `pip install` referencing a git repo/branch directly but this seemed safer.
